test/gtest/conv_embed_db.cpp

```diff
@@ -0,0 +1,229 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <tuple>
+
+#include <miopen/miopen.h>
+#include <gtest/gtest.h>
+#include <miopen/miopen.h>
+#include <miopen/env.hpp>
+#include "conv_2d.hpp"
+#include "get_handle.hpp"
+
+void GetArgs(const std::string& param, std::vector<std::string>& tokens)
+{
+    std::stringstream ss(param);
+    std::istream_iterator<std::string> begin(ss);
+    std::istream_iterator<std::string> end;
+    while(begin != end)
+        tokens.push_back(*begin++);
+}
+
+class ConfigWithHalf : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+class ConfigWithInt8 : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+class ConfigWithBFloat16 : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+class ConfigWithFloat : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+
+void Run2dDriver(miopenDataType_t prec)
+{
+
+    std::vector<std::string> params;
+    switch(prec)
+    {
+    case miopenFloat: params = ConfigWithFloat::GetParam(); break;
+    case miopenHalf: params = ConfigWithHalf::GetParam(); break;
+    case miopenInt8: params = ConfigWithInt8::GetParam(); break;
+    case miopenBFloat16: params = ConfigWithBFloat16::GetParam(); break;
+    case miopenInt8x4:
+    case miopenInt32:
+    case miopenDouble:
+        MIOPEN_THROW(miopenStatusBadParm,
+                     "miopenInt8x4, miopenInt32, miopenDouble data type not supported by "
+                     "conv_embed_db test");
+
+    default: params = ConfigWithFloat::GetParam();
+    }
+
+    for(const auto& test_value : params)
+    {
+        std::vector<std::string> tokens;
+        GetArgs(test_value, tokens);
+        std::vector<const char*> ptrs;
+
+        std::transform(tokens.begin(), tokens.end(), std::back_inserter(ptrs), [](const auto& str) {
+            return str.data();
+        });
+
+        testing::internal::CaptureStderr();
+        test_drive<conv2d_driver>(ptrs.size(), ptrs.data());
+        auto capture = testing::internal::GetCapturedStderr();
+        EXPECT_FALSE(capture.find("Perf Db: record not found") != std::string::npos);
+    }
+};
+
+bool IsTestSupportedForDevice(const miopen::Handle& handle)
+{
+    std::string devName = handle.GetDeviceName();
+    if(devName == "gfx900" || devName == "gfx906")
+        return true;
+    else
+        return false;
+}
+
+TEST_P(ConfigWithFloat, FloatTest)
+{
+#if MIOPEN_EMBED_DB
+
+    const auto& handle = get_handle();
+    const auto& envVar = miopen::GetEnv("MIOPEN_TEST_FLOAT_ARG");
```

Comment:
Please use miopen::GetStringEnv() as it implements caching. Example:
```C++
MIOPEN_DECLARE_ENV_VAR(MIOPEN_TEST_FLOAT_ARG)
...
static std::string GetFloatArg()
{
    const char* p = miopen::GetStringEnv(MIOPEN_TEST_FLOAT_ARG{});
    if(p == nullptr)
    {
        // Fail the test (preferred) or return "--float" by default.
    }
    return {p};
}
```

Reply:
This matches the proven pattern that we use in MIOpen when we need to read a string.

Reply:
[Resolved]